app/lib/frontend/handlers/documentation.dart

```diff
@@ -43,6 +45,39 @@ Future<shelf.Response> documentationHandler(shelf.Request request) async {
     return redirectResponse('${request.requestedUri}/');
   }
   final String requestMethod = request.method.toUpperCase();
+
+  if (requestContext.experimentalFlags.showSandboxedOutput) {
+    if (requestMethod != 'HEAD' && requestMethod != 'GET') {
```

Comment:
The existing code also does this. This is cargo culting.

We could throw `throw AssertionError` instead, I think it's reasonable to check that the method is as expected.

But yes, we could have an invariant that `documentationHandler` only gets called for `GET` and `HEAD` requests.
I think it's safer to say that an `xxxxHandler` function should generally check the method and return method-not-supported, response if the method isn'ẗ supported.
It's generally nicer than making it an internal error, and it's almost always correct :D

-------------

But yes, we can choose:
 * (A) `xxxxHandler` only supports some methods, and violating this is an internal error.
    * So we throw `AssertionError` if called with method not allowed (or `UnsupportedError`) -- In either case, it's an internal error.
    * The caller is responsible for checking the method (in this case `router.dart` does that).
    * Documentation comment must specify the supported methods -- I'm not sure we're good at writing these.
 * (B)  `xxxxHandler` checks the `request.method` and returns an method-not-supported error.
    * We can probably make this fairly simple by creating a `ResponseException` for method-not-supported. Then we just have `MethodNotSupportedException.checkGetOrHead(request)` function to call.

I think that in general it would be safer to go with (B), and that we should do this in a separate PR, I've filed https://github.com/dart-lang/pub-dev/issues/6205